km/km_hcalls.c

```diff
@@ -1139,17 +1139,10 @@ static km_hc_ret_t rt_sigprocmask_hcall(void* vcpu, int hc, km_hc_args_t* arg)
 static km_hc_ret_t rt_sigaction_hcall(void* vcpu, int hc, km_hc_args_t* arg)
 {
    // int rt_sigaction(int signum, const struct sigaction *act, struct sigaction *oldact)
-   km_sigaction_t* act = NULL;
-   km_sigaction_t* oldact = NULL;
+   // note: act or oldact can be NULL. Let the lower level function sort it out.
+   km_sigaction_t* act = km_gva_to_kma(arg->arg2);
+   km_sigaction_t* oldact = km_gva_to_kma(arg->arg3);
 
-   if (arg->arg2 != 0 && (act = km_gva_to_kma(arg->arg2)) == NULL) {
-      arg->hc_ret = -EFAULT;
-      return HC_CONTINUE;
-   }
-   if (arg->arg3 != 0 && (oldact = km_gva_to_kma(arg->arg3)) == NULL) {
-      arg->hc_ret = -EFAULT;
-      return HC_CONTINUE;
-   }
```

Comment:
Please don't remove this.
With this change if arg2 or arg3 are an invalid payload address km will pass NULL to the kernel with this system call.
The kernel will see the NULL and think the caller didn't supply these arguments and perform the system call as though the arguments were not supplied when the call should have failed with error EFAULT.